src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/ResolveTargetingPackAssetsTests.cs

```diff
@@ -12,15 +16,82 @@ public class ResolveTargetingPackAssetsTests
         [Fact]
         public void Given_ResolvedTargetingPacks_with_valid_PATH_in_PlatformManifest_It_resolves_TargetingPack()
         {
-            string mockPackageDirectory = Path.Combine(Path.GetTempPath(), "dotnetSdkTests", Path.GetRandomFileName());
+            ResolveTargetingPackAssets task = InitializeTask(out string mockPackageDirectory);
+
+            task.Execute().Should().BeTrue();
+
+            task.ReferencesToAdd[0].ItemSpec.Should().Be(Path.Combine(mockPackageDirectory, "lib/Microsoft.Windows.SDK.NET.dll"));
+            task.PlatformManifests[0].ItemSpec.Should().Be(Path.Combine(mockPackageDirectory, $"data{Path.DirectorySeparatorChar}PlatformManifest.txt"));
+            task.AnalyzersToAdd.Length.Should().Be(2);
+            task.AnalyzersToAdd[0].ItemSpec.Should().Be(Path.Combine(mockPackageDirectory, "analyzers/dotnet/anyAnalyzer.dll"));
+            task.AnalyzersToAdd[1].ItemSpec.Should().Be(Path.Combine(mockPackageDirectory, "analyzers/dotnet/cs/csAnalyzer.dll"));
+
+            ((MockBuildEngine4)task.BuildEngine).RegisteredTaskObjectsQueries.Should().Be(2,
+                because: "There should be a lookup for the overall and the specific targeting pack");
+
+            ((MockBuildEngine4)task.BuildEngine).RegisteredTaskObjects.Count.Should().Be(2,
+                because: "There should be a cache entry for the overall lookup and for the specific targeting pack");
+        }
+
+        [Fact]
+        public void Given_Passing_ResolvedTargetingPacks_It_Passes_Again_With_Cached_Results()
+        {
+            ResolveTargetingPackAssets task = InitializeTask(out string _);
+
+            task.Execute().Should().BeTrue();
+
+            ((MockBuildEngine4)task.BuildEngine).RegisteredTaskObjectsQueries.Should().Be(2,
+                because: "there should be a lookup for the overall and the specific targeting pack");
+
+            ((MockBuildEngine4)task.BuildEngine).RegisteredTaskObjects.Count.Should().Be(2,
+                because: "there should be a cache entry for the overall lookup and for the specific targeting pack");
+
+            task.Execute().Should().BeTrue();
```

Comment:
Should we be creating a new task instance here instead of reusing the existing one?